gpii/node_modules/userListeners/src/listeners.js

```diff
@@ -208,3 +229,78 @@ gpii.userListeners.failed = function (that, err) {
         });
     });
 };
+
+/**
+ * Reads a token from the file.
+ *
+ * @param {String} tokenDirectory - The directory containing the token file.
+ * @param {String} tokenFile [optional] The file, in tokenDirectory, containing the token. Can be omitted if the first
+ *   argument is the full path to the file.
+ * @return {Promise} resolves when the token is read, rejects if there's no file (or there was an error reading it).
+ */
+gpii.userListeners.readTokenFile = function (tokenDirectory, tokenFile) {
+    var promise = fluid.promise();
+
+    // Try to read the token file.
+    var tokenPath = path.join(tokenDirectory, tokenFile || "");
+    fluid.log("Reading token file: ", tokenPath);
+    fs.readFile(tokenPath, "utf8", function (err, data) {
+        if (err) {
+            if (err.code === "ENOENT") {
+                fluid.log("Token file not found");
+                promise.reject();
+            } else {
+                var message = "Error reading token file " + tokenPath + ": " + err.message;
+                fluid.log(message);
+                promise.reject({
+                    isError: true,
+                    message: message,
+                    error: err
+                });
+            }
+        } else {
+            var token = data.trim();
+            if (token.length > 0) {
+                fluid.log("Got token: ", token);
+                promise.resolve(token);
+            } else {
+                fluid.log("Empty token file");
+                promise.reject();
+            }
+        }
+    });
+
+    return promise;
+};
+
+/**
+ * Writes a token to a file.
+ *
+ * @param {String} tokenDirectory - The directory containing the token file.
+ * @param {String} tokenFile [optional] The file, in tokenDirectory, containing the token. Can be omitted if the first
+ *   argument is the full path to the file.
+ * @param {String} tokenValue - The token's value.
+ * @return {Promise} resolves when complete, with a value of true if it was written.
+ */
+gpii.userListeners.writeTokenFile = function (tokenDirectory, tokenFile, tokenValue) {
+    var promise = fluid.promise();
+
+    var tokenPath = path.join(tokenDirectory, tokenFile || "");
+    fluid.log("Writing token '" + tokenValue + "' to ", tokenPath);
+
+    fs.writeFile(tokenPath, tokenValue, "utf8", function (err) {
+        if (err) {
+            var message = "Error writing token file " + tokenPath + ": " + err.message;
+            fluid.log(message);
+            promise.reject({
+                isError: true,
+                message: message,
+                error: err
```

Comment:
According to the convention, the path for the original error object is named as "originalError" rather than "error".